packages/typescript-estree/src/create-program/getWatchProgramsForProjects.ts

```diff
@@ -270,6 +270,7 @@ function createWatchProgram(
   ) as WatchCompilerHostOfConfigFile<ts.BuilderProgram>;
 
   if (parseSettings.moduleResolver) {
+    // eslint-disable-next-line deprecation/deprecation -- intentional for older TS versions
     watchCompilerHost.resolveModuleNames = getModuleResolver(
```

Comment:
`resolveModuleName` is deprecated with the following message:

```
supply resolveModuleNameLiterals instead for resolution that can handle newer resolution modes like nodenext

If provided, used to resolve the module names, otherwise typescript's default module resolution
```

Reply:
Should we add new options for `resolveModuleNameLiterals`?

Reply:
We'll need to figure out a solution for this, yeah.
Perhaps we need to deprecate this featured for v5+ instead?
Idk - this feature wasn't ever well planned/thought out and does have some known problems with it.

Reply:
> Perhaps we need to deprecate this featured for v5+ instead?

Maybe yes. I'll create new PR to deprecate it for v5+.

Reply:
We'll drop this in v6
https://github.com/typescript-eslint/typescript-eslint/issues/3798#issuecomment-1465092927